pkg/security/tests/security_profile_test.go

```diff
@@ -0,0 +1,506 @@
+						return true
+					}
+					return false
+				})
+				if nodes == nil {
+					t.Fatal("Node not found in security profile")
+				}
+				if len(nodes) != 1 {
+					t.Fatalf("Found %d nodes, expected only one.", len(nodes))
+				}
+				for name := range nodes[0].DNSNames {
+					if name == "foo.bar" {
+						return true
+					}
+				}
+				t.Error("DNS req not found in security profile")
+				return false
+			})
+	})
+}
+
+func TestAnomalyDetection(t *testing.T) {
+	// skip test that are about to be run on docker (to avoid trying spawning docker in docker)
+	if testEnvironment == DockerEnvironment {
+		t.Skip("Skip test spawning docker containers on docker")
+	}
+	if _, err := whichNonFatal("docker"); err != nil {
+		t.Skip("Skip test where docker is unavailable")
+	}
+	if !IsDedicatedNode(dedicatedADNodeForTestsEnv) {
+		t.Skip("Skip test when not run in dedicated env")
+	}
+
+	var expectedFormats = []string{"profile"}
+	var testActivityDumpTracedEventTypes = []string{"exec", "open", "syscalls", "dns", "bind"}
+
+	outputDir := t.TempDir()
+	os.MkdirAll(outputDir, 0755)
+	defer os.RemoveAll(outputDir)
+	test, err := newTestModule(t, nil, []*rules.RuleDefinition{}, testOpts{
+		enableActivityDump:                  true,
+		activityDumpRateLimiter:             200,
+		activityDumpTracedCgroupsCount:      3,
+		activityDumpCgroupDumpTimeout:       10,
+		activityDumpLocalStorageDirectory:   outputDir,
+		activityDumpLocalStorageCompression: false,
+		activityDumpLocalStorageFormats:     expectedFormats,
+		activityDumpTracedEventTypes:        testActivityDumpTracedEventTypes,
+		enableSecurityProfile:               true,
+		securityProfileDir:                  outputDir,
+		securityProfileWatchDir:             true,
+		anomalyDetectionMinimumStablePeriod: 0,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer test.Close()
+	syscallTester, err := loadSyscallTester(t, test, "syscall_tester")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("anomaly-detection-process", func(t *testing.T) {
+		dockerInstance, dump, err := test.StartADockerGetDump()
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer dockerInstance.stop()
+
+		time.Sleep(time.Second * 1) // to ensure we did not get ratelimited
+		cmd := dockerInstance.Command(syscallTester, []string{"sleep", "1"}, []string{})
+		_, err = cmd.CombinedOutput()
+		if err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(1 * time.Second) // a quick sleep to let events to be added to the dump
+
+		err = test.StopActivityDump(dump.Name, "", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(6 * time.Second) // a quick sleep to let the profile to be loaded (5sec debounce + 1sec spare)
+
+		err = test.GetCustomEventSent(t, func() error {
+			cmd := dockerInstance.Command("getconf", []string{"-a"}, []string{})
+			_, err = cmd.CombinedOutput()
+			return err
+		}, func(r *rules.Rule, event *events.CustomEvent) bool {
+			assert.Equal(t, events.AnomalyDetectionRuleID, r.Rule.ID, "wrong custom event rule ID")
```

Comment:
I would check that this is the expected anomaly detection event for this process or add a negative test

Reply:
I agree it's far from perfect here, sadly the custom event handler only provide the rule, eventType, tags and marshalerCtor. This will require a rework to be able to write better tests.
Meanwhile I'll add some negative tests, good idea, thanks!